skippkg-finder.py

```diff
@@ -0,0 +1,333 @@
+                    else:
+                        logging.info("Can not find binary of %s" % index)
+
+        # a list of binary RPM should filter out from ftp
+        obsoleted = []
+        for pkg in fullbinarylist:
+            if pkg not in selected_binarylist and pkg not in obsoleted and not self.exceptions(pkg):
+                # special handling for -release package
+                if pkg == 'openSUSE-release' or pkg == 'openSUSE-release-ftp' or\
+                        pkg == 'openSUSE-Addon-NonOss-release':
+                    continue
+                obsoleted.append(pkg)
+
+        # another ugly hack for -32bit and -64bit binary RPM for the obsoleted list
+        unneeded = obsoleted.copy()
+        for pkg in unneeded:
+            if pkg.endswith('-32bit') or pkg.endswith('-64bit'):
+                main_filename = re.sub('-[36][24]bit', '', pkg)
+                if main_filename not in obsoleted:
+                    obsoleted.remove(pkg)
+
+        skip_list = ET.Element('group', {'name': 'NON_FTP_PACKAGES'})
+        ET.SubElement(skip_list, 'conditional', {'name': 'drop_from_ftp'})
+        packagelist = ET.SubElement(skip_list, 'packagelist', {'relationship': 'requires'})
+        for pkg in sorted(obsoleted):
+            if not self.print_only and self.verbose:
+                print(pkg)
+            attr = {'name': pkg}
+            ET.SubElement(packagelist, 'package', attr)
+        if not self.print_only:
+            source_file_ensure(self.apiurl, self.opensuse_project, META_PACKAGE, 'NON_FTP_PACKAGES.group',
+                               ET.tostring(skip_list, pretty_print=True, encoding='unicode'),
+                               'Update the skip list')
+        else:
+            print(ET.tostring(skip_list, pretty_print=True,
+                  encoding='unicode'))
+
+
+def main(args):
+    osc.conf.get_config(override_apiurl=args.apiurl)
+    osc.conf.config['debug'] = args.debug
+
+    uc = SkippkgFinder(args.opensuse_project, args.sle_project, args.print_only, args.verbose)
+    uc.crawl()
+
+
+if __name__ == '__main__':
+    description = 'Overwrites NON_FTP_PACKAGES.group according to the latest sources. '\
+                  'This tool only works for Leap after CtLG implemented.'
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument('-A', '--apiurl', metavar='URL', help='API URL')
+    parser.add_argument('-d', '--debug', action='store_true',
+                        help='print info useful for debuging')
+    parser.add_argument('-o', '--opensuse-project', dest='opensuse_project', metavar='OPENSUSE_PROJECT',
+                        help='openSUSE project on buildservice (default: %s)' % OPENSUSE,
+                        default=OPENSUSE)
+    parser.add_argument('-s', '--sle-project', dest='sle_project', metavar='SLE_PROJECT',
+                        help='SLE project on buildservice (default: %s)' % SLE,
+                        default=SLE)
```

Comment:
I am not sure if those defaults are good idea as they become obsolete in next release, so we either break compatibility with updated default or has obsolete default.